addons/account/wizard/account_move_send.py

```diff
@@ -595,7 +595,11 @@ def action_send_and_print(self, from_cron=False, allow_fallback_pdf=False):
             # Manage errors.
             errors = {move: move_data for move, move_data in moves_data.items() if move_data.get('error')}
             if errors:
-                self._hook_if_errors(errors, from_cron=from_cron, allow_fallback_pdf=allow_fallback_pdf)
+                try:
```

Comment:
Maybe I should add the try-except around the whole generate_invoice_documents block since there are multiple sub-methods called inside and in the future they could be source of "raise" ?

Reply:
@clbr-odoo Currently, nothing should raise outside of `_hook_if_errors`. Otherwise, it could break the existing flow. But I agree with you. The try/except should be on the whole method. However, setting `self.mode = 'done'` is nice in case of error but you should recompute the "correct" mode at the beginning of the method in case of the user retries with different options